app/assets/javascripts/discourse/app/services/admin-custom-user-fields.js

```diff
@@ -0,0 +1,13 @@
+import { tracked } from "@glimmer/tracking";
+import Service from "@ember/service";
+
+export default class AdminCustomUserFields extends Service {
+  @tracked additionalProperties = [];
+
+  addProperties(properties) {
+    if (typeof properties === "string") {
+      properties = [properties];
+    }
+    this.additionalProperties.push(...properties);
```

Comment:
I personally prefer a fn that only allows one property at a time, and you call it N times if you want to add N fields, rather than mixing arrays and strings... but I'll let you decide, I don't think it's a hard rule